robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java

```diff
@@ -682,16 +688,56 @@ private String createTempDir(String name) {
         .toString();
   }
 
+  private static BroadcastReceiver newBroadcastReceiverFromP(
+      String receiverClassName,
+      Application application,
+      LoadedApk loadedApk,
+      BroadcastReceiverData receiver) {
+    ClassLoader classLoader;
```

Comment:
You don't need all of this classloader logic, there is only one classloader that works (the Robolectric one). You can just use `Shadow.class.getClassLoader()`.